e2e/preferences.e2e.spec.ts

```diff
@@ -0,0 +1,142 @@
+import os from 'os';
+import path from 'path';
+import { ElectronApplication, BrowserContext, _electron, Page } from 'playwright';
+import { test, expect } from '@playwright/test';
+import { createDefaultSettings, packageLogs, reportAsset } from './utils/TestUtils';
+import { NavPage } from './pages/nav-page';
+import { PreferencesPage } from './pages/preferences';
+
+let page: Page;
+
+/**
+ * Using test.describe.serial make the test execute step by step, as described on each `test()` order
+ * Playwright executes test in parallel by default and it will not work for our app backend loading process.
+ * */
+test.describe.serial('Main App Test', () => {
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+  let preferencesWindow: Page;
+
+  test.beforeAll(async() => {
+    createDefaultSettings();
+
+    electronApp = await _electron.launch({
+      args: [
+        path.join(__dirname, '../'),
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        // See src/utils/commandLine.ts before changing the next item as the final option.
+        '--disable-dev-shm-usage',
+        '--no-modal-dialogs',
+      ],
+      env: {
+        ...process.env,
+        RD_LOGS_DIR: reportAsset(__filename, 'log'),
+      },
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+
+    new NavPage(page).preferencesButton.click();
+
+    preferencesWindow = await electronApp.waitForEvent('window', page => /preferences/i.test(page.url()));
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: reportAsset(__filename) });
+    await packageLogs(__filename);
+    await electronApp.close();
+  });
+
+  test('should open preferences modal', () => {
+    expect(preferencesWindow).toBeDefined();
+  });
+
+  test('should show application page', async() => {
+    const { application } = new PreferencesPage(preferencesWindow);
+
+    await expect(application.nav).toHaveClass('preferences-nav-item active');
+
+    if (!os.platform().startsWith('win')) {
+      await expect(application.tabBehavior).toHaveText('Behavior');
+      await expect(application.administrativeAccess).toBeVisible();
+    }
+
+    await expect(application.automaticUpdates).toBeVisible();
+    await expect(application.statistics).toBeVisible();
+    await expect(application.pathManagement).not.toBeVisible();
+  });
+
+  test('should render environment tab', async() => {
+    test.skip(os.platform() === 'win32', 'Environment tab not available on Windows');
+    const { application } = new PreferencesPage(preferencesWindow);
+
+    application.tabEnvironment.click();
```

Comment:
All `click()`s should be `await`ed upon, to make sure that the processing has completed (we've loaded the new tab).